File: 프로그래머스/unrated/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.js
solution = (numbers) => {
    return Array(numbers.length)
        .fill(-1)
        .reduce((stack, el, idx, arr) => {
            while (stack && numbers[stack.at(-1)] < numbers[idx])
                arr[stack.pop()] = numbers[idx];
            stack.push(idx);
            return idx === numbers.length - 1 ? arr : stack;
        }, []);
};

File: 프로그래머스/unrated/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

using p = pair<int, int>;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    stack<p> st;

    answer.resize(numbers.size());

    for (int i = 0; i < numbers.size(); ++i)
    {
        while (!st.empty() && st.top().second < numbers[i])
        {
            answer[st.top().first] = numbers[i];
            st.pop();
        }
        st.push(make_pair(i, numbers[i]));
    }

    while (!st.empty())
    {
        answer[st.top().first] = -1;
        st.pop();
    }

    return answer;
}   